runtime/CMakeLists.txt

```diff
@@ -49,6 +49,13 @@ FetchContent_Declare(
     SOURCE_DIR          mlir/ExecutionEngine
 )
 
+FetchContent_Declare(
+    MLIRFloat16Bits
+    URL                 https://raw.githubusercontent.com/llvm/llvm-project/${LLVM_HASH}/mlir/include/mlir/ExecutionEngine/Float16bits.h
+    DOWNLOAD_NO_EXTRACT True
+    SOURCE_DIR          mlir/ExecutionEngine
+)
```

Comment:
Is this not available in the version of MLIR that we are using locally? Also, do we need it for anything?

Reply:
It's required by RunnerUtils.h from MLIR. They updateed RunnerUtils.h with 16-bit functions ([commit link](https://github.com/llvm/llvm-project/commit/7bc6c4abe8e8d8ab70e02e4c2025a94dda01d908)) and now requires the library.

(CMAKE RunnerUtilh)
https://github.com/PennyLaneAI/catalyst/blob/d5b8419c6ec608878ae5d1baab7987b90dacbfd4/runtime/CMakeLists.txt#L38-L42

(RunnerUtils.h needs Float16Bits)
https://github.com/llvm/llvm-project/blob/7bc6c4abe8e8d8ab70e02e4c2025a94dda01d908/mlir/include/mlir/ExecutionEngine/RunnerUtils.h#L40-L43

Reply:
I see! Thanks. But wouldn't this file exist in our LLVM submodule and not require its download?

Reply:
Yes it's there. I'll try importing it instead of downloading.

Reply:
We download certain mlir headers via cmake for the runtime to avoid having to clone the entire llvm repository (which we normally do to build the compiler) when building the runtime.

Reply:
Yes, I saw later with Tzung-han and mentioned that it is ok if we fetch. My bad.

Reply:
(But just FYI, we don't need to clone the entire LLVM repo, if we just copy the file instead of downloading it. But it is fine...)

Reply:
Copying means it goes out of date no?

Reply:
Not necessarily. Copying/Moving at the time from CMake should be fine. I think it would be equivalent to what we have here.